### ISSUE TITLE
Sync of shared library stops working, if ownership changes

### ISSUE THREAD

**Issue by egroeper** [haiwen/seafile-client#937]
We recently got a couple of complaints, when Seafile is used in larger groups and one or more of the libraries shared to a group get transferred to a new user.
In the Webinterface everything still works fine, but the clients stop syncing the library.
This of course causes big problems, because files get out of sync, we get a couple of different file versions, that then have to be merged manually.
Of course one of the problems here is, that users don't stop working on the library after the error occurs.

But the error could really be fixed automatically, since everything works fine, when resyncing the library (which the error message suggests; but not in a way the users understand, if the german translation is used).

So please: Why does the library need to stop syncing in such cases? Isn't it possible to automatically recheck permissions for the library, when the error is detected?
Perhaps only a couple of times (at least one time), if it's not possible to clearly distinguish between this situation and other errors.

This is kind of related to #655.

For sake of completeness here the relevant part of the seafile.log:

```
[06/26/17 13:59:36] sync-mgr.c(702): Repo 'Test_Ownership_Change' sync state transition from 'synchronized' to 'uploading'.
[06/26/17 13:59:36] http-tx-mgr.c(3423): Upload with HTTP sync protocol version 1.
[06/26/17 13:59:36] http-tx-mgr.c(1132): Transfer repo '505a7220': ('normal', 'init') --> ('normal', 'check')
[06/26/17 13:59:36] http-tx-mgr.c(2255): Bad response code for GET https://filesync-test.ostack.hu-berlin.de/seafhttp/repo/505a7220-b429-4bd9-a4c9-134ca89536e3/permission-check/?op=upload&client_id=ec252d57c33a920a71dcd3bbcd6bcb8d8ad19e91&client_name=penny: 403.
[06/26/17 13:59:36] http-tx-mgr.c(3441): Upload permission denied for repo 505a7220 on server https://filesync-test.ostack.hu-berlin.de.
[06/26/17 13:59:36] http-tx-mgr.c(1132): Transfer repo '505a7220': ('normal', 'check') --> ('error', 'finished')
[06/26/17 13:59:36] sync-mgr.c(764): Repo 'Test_Ownership_Change' sync state transition from uploading to 'error': 'You do not have permission to access this library'.
```

Unfortunately I found a follow-up error of this, which is more serious. Will post that in another ticket.

**Comment by shoeper** [comment]
Settings -> Advanced -> do not unsync if not found on server could likely work around it.

Looking at the log it looks like when the owner changes the sharing at least gets lost temporarly which leads to the client to unsync it because it thinks the library is not there anymore.

**Comment by egroeper** [comment]
@shoeper: Where do you see that? The library is not unsynced. It has a problem with syncing. It's in sync-error state. For example in the gui it shows an exclamation mark, because there is a problem. It doesn't show the cloud symbol for unsynced libraries.

**Comment by shoeper** [comment]
Oh, I thought that happens when a permissions error occurs

```
Upload permission denied for repo 505a7220 on server https://filesync-test.ostack.hu-berlin.de.
```

What happens after restarting the client? Does it sync again?

**Comment by egroeper** [comment]
No. Restart doesn't help. Resync ("Neu synchronisieren") helps, as stated above.

**Comment by jobisoft** [comment]
I can confirm this bug!

We used the "transfer" option in the web interface to move a shared folder from one user to another. After the transfer, everything looked normal in the web interface: The owner of the group did not change, just the user which owns the folder changed ("shared by") as intended.

All group members could continue to access the folder via the web interface, but all users using the sync desktop client (6.0.7) got a red "x" attached to the folder icon in windows explorer and a big error in the folder list of the seafile client. The only way to fix this was to terminate/disconnect the syncing of those folders ("Synchronisation trennen") and sync again ("Herunterladen und synchronisieren").

**Comment by jobisoft** [comment]
Maybe it is important: The client server connection is using shibboleth.

**Comment by killing** [comment]
Since recent versions (12 or so), you can choose to keep share relationships when transferring a library.
